### PR TITLE
use deepchecks metrics for precision recall f1

### PR DESCRIPTION
**Pull request by noamzbr**
#### Reference Issues/PRs

Closes https://github.com/deepchecks/private-issues/issues/73 (For now, we'll probably want to keep improving and extend improvements to other metrics)

#### What does this implement/fix? Explain your changes.
Uses the metrics implemented in https://github.com/deepchecks/deepchecks-metrics to reduce the runtime of precision/recall/f1 by around %60

Does not cover regression metrics and roc_auc for example.

### CODE REVIEW

**Review by ItayGabbay — APPROVED** [review]

File: deepchecks/tabular/checks/model_evaluation/single_dataset_performance.py [hunk 1]
```diff
@@ -56,7 +56,10 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         results = []
         classes = dataset.classes
         label = cast(pd.Series, dataset.label_col)
-        n_samples = label.groupby(label).count()
+        if context.with_display:
+            n_samples = label.groupby(label).count()
+        else:
+            n_samples = dict(zip(classes, [None] * len(classes)))
```
Comment (Nadav-Barak) [inline deepchecks/tabular/checks/model_evaluation/single_dataset_performance.py:62]: weird that n_samples is a dict, consider changing the name

File: deepchecks/tabular/checks/model_evaluation/single_dataset_performance.py [hunk 2]
```diff
@@ -56,7 +56,10 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         results = []
         classes = dataset.classes
         label = cast(pd.Series, dataset.label_col)
-        n_samples = label.groupby(label).count()
+        if context.with_display:
+            n_samples = label.groupby(label).count()
```
Comment (Nadav-Barak) [inline deepchecks/tabular/checks/model_evaluation/single_dataset_performance.py:60]: can use Counter

File: deepchecks/tabular/metric_utils/scorers.py [hunk 3]
```diff
@@ -105,9 +109,16 @@
 multiclass_scorers_dict = {
     'accuracy': get_scorer('accuracy'),
```
Comment (Nadav-Barak) [inline deepchecks/tabular/metric_utils/scorers.py:110]: Loved how you kept the order

**Comment by Nadav-Barak** [comment]
Also, https://github.com/deepchecks/deepchecks-metrics does not exist

File: deepchecks/tabular/checks/model_evaluation/single_dataset_performance.py (hunk 1 above)
Comment (noamzbr) [reply deepchecks/tabular/checks/model_evaluation/single_dataset_performance.py:62]: It's a series in the with_display==True case, but a dict has the same api

File: deepchecks/tabular/checks/model_evaluation/single_dataset_performance.py (hunk 2 above)
Comment (noamzbr) [reply deepchecks/tabular/checks/model_evaluation/single_dataset_performance.py:60]: if it's the with_display case then efficiency isn't that important

**Pull request merged**